src/map/entities/charentity.cpp

```diff
@@ -2148,7 +2148,7 @@ void CCharEntity::Die()
 
     battleutils::RelinquishClaim(this);
 
-    if (this->PPet)
+    if (this->PPet && this->objtype == TYPE_PC)
```

Comment:
Is this required?  CCharEntity can't be cast to anything that wouldn't have objtype == TYPE_PC.  If it's a mob with a pet, it would have to be a CMobEntity...

Reply:
I thought these two would coincide. Ever since this fix was made, this bug arose lol. But perhaps you're right.

Reply:
I think it's related to the MobEntity::Die() function further calling PPet->Die(), but PPet is a CBattleEntity.  I'm wonding if it needs to be re-cast as a mob or pet entity perhaps?  The CPetEntity::Die() function calls back to CBattleEntity() but clears other stuff first.  Would probably need more testing.

Reply:
So I realized this is entirely the improper way to go about this. this bug I'm referring to is only affecting mobs that despawn during day time. AKA fomor.
If a BST dies, its pet stays alive. SMN and PUP's pets definitely die, but I'm not too sure about DRG and their wyvern.

Reply:
@paladindamarus Most recent commit fixes this appropriately :)

Reply:
> 

Wyverns and BST pets stay up.